src/many-sources/abstract-pool.js

```diff
@@ -34,8 +35,15 @@ inherit(AbstractPool, Stream, {
       if (this._queueLim === -1 || this._queue.length < this._queueLim) {
         this._addToQueue(toObs(obj));
       } else if (this._drop === 'old') {
-        this._removeOldest();
-        this._add(obj, toObs);
+        if (this._overlapping) {
+          this._addToCur(toObs(obj));
+          while (this._curSources.length > this._concurLim) {
+            this._removeOldest();
+          }
```

Comment:
Ah, right! I didn't realize that `_removeOldest` pulls from the `queue`. Replacing `this._add -> this._addToCur` isn't equivalent then indeed. I'll think about it too.